prog/vm/github_runner.rb

```diff
@@ -188,14 +189,24 @@ def before_run
   end
 
   label def wait
+    # STATE_MAP is defined in rhizome/common/bin/daemonizer script.
+    #   dead    => NotStarted
+    #   running => InProgress
+    #   exited  => Succeeded
+    #   failed  => Failed
     case vm.sshable.cmd("systemctl show -p SubState --value #{SERVICE_NAME}").chomp
     when "exited"
       github_runner.incr_destroy
       nap 0
     when "failed"
+      Clog.emit("#{SERVICE_NAME} is failed, registering the runner again") { {github_runner: github_runner.values} }
+      vm.sshable.cmd("sudo systemctl stop runner-script")
```

Comment:
For failures, you need to use `sudo systemctl reset-failed runner-script.service`. (see the daemonizer)

Also we use SERVICE_NAME in some places and runner-script in some others. I think you don't need SERVICE_NAME constant. You can use runner-script directly.

Reply:
Changed it